rego-templates/embedded.go

```diff
@@ -0,0 +1,101 @@
+package rego_templates
+
+import (
+	"bufio"
+	"embed"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/aquasecurity/postee/v2/data"
+	"github.com/aquasecurity/postee/v2/log"
+)
+
+const (
+	COMMON_DIR               = "common"
+	LOCAL_DIR                = "."
+	REGO_PACKAGE_DECLERATION = "package"
+)
+
+var (
+	//go:embed *.rego
+	EmbeddedFiles embed.FS
+	templates     = make(map[string]string)
+	templatesMu   sync.Mutex
+
+	//go:embed common
+	EmbeddedCommonFiles embed.FS
+	commonTemplates     = make(map[string]string)
+	commonMu            sync.Mutex
+)
+
+func EmbeddedTemplates() map[string]string {
+	templatesMu.Lock()
+	defer templatesMu.Unlock()
+	if len(templates) != 0 {
+		return templates
+	}
+	populateFS(EmbeddedFiles, templates, LOCAL_DIR)
+
+	return templates
+}
+
+func EmbeddedCommon() map[string]string {
+	commonMu.Lock()
+	defer commonMu.Unlock()
+	if len(commonTemplates) != 0 {
+		return commonTemplates
+	}
+	populateFS(EmbeddedCommonFiles, commonTemplates, COMMON_DIR)
+
+	return commonTemplates
+}
+
+func populateFS(files embed.FS, storage map[string]string, dirPath string) {
```

Comment:
I would define it as `populateFS(files embed.FS, dirPath string) map[string]string`. This way purpose of function is more clear

Reply:
I re-read it again and see name is a bit incorrect. It's called `populateFS()` but `FS` instance isn't modified and just used as source to build a `map[string] string`